dm/master/election.go

```diff
@@ -55,21 +57,51 @@ func (s *Server) electionNotify(ctx context.Context) {
 				continue
 			}
 
+			failToStart := false
+
 			if leaderInfo.ID == s.cfg.Name {
 				// this member become leader
 				log.L().Info("current member become the leader", zap.String("current member", s.cfg.Name))
 				err := s.scheduler.Start(ctx, s.etcdClient)
 				if err != nil {
 					log.L().Error("scheduler do not started", zap.Error(err))
+					failToStart = true
 				}
 
-				err = s.pessimist.Start(ctx, s.etcdClient)
-				if err != nil {
-					log.L().Error("pessimist do not started", zap.Error(err))
+				if !failToStart {
+					err = s.pessimist.Start(ctx, s.etcdClient)
+					if err != nil {
+						log.L().Error("pessimist do not started", zap.Error(err))
+						failToStart = true
+					}
 				}
-				err = s.optimist.Start(ctx, s.etcdClient)
-				if err != nil {
-					log.L().Error("optimist do not started", zap.Error(err))
+
+				if !failToStart {
+					err = s.optimist.Start(ctx, s.etcdClient)
+					if err != nil {
+						log.L().Error("optimist do not started", zap.Error(err))
+						failToStart = true
+					}
+				}
```

Comment:
How about put these into a function? It looks clearer to me.

Reply:
change in 8e98233